dashboard/dashboard_script.py

```diff
@@ -33,21 +39,41 @@ def fetch_data_from_tables(conn):
         demand_query = "SELECT * FROM Demand"
         gen_query = "SELECT * FROM Generation"
         carbon_query = "SELECT * FROM Carbon"
+        piechart_query = "SELECT * FROM generation_percent"
         demand_df = pd.read_sql_query(demand_query, conn)
         gen_df = pd.read_sql_query(gen_query, conn)
         carbon_df = pd.read_sql_query(carbon_query, conn)
+        carbon_df = carbon_df[-48:]
+        pie_df = pd.read_sql_query(piechart_query, conn)
 
         if conn:
             conn.close()
-            return gen_df, demand_df,   carbon_df
+            return gen_df, demand_df,   carbon_df, pie_df
         raise ConnectionError("Failed to connect to Database")
 
+    def submit_form(self, name, email, postcode, hours_to_charge, charging_preference):
```

Comment:
Is this susceptible to an injection attacks? Okay got MVP but will need altering.